frame/offences/benchmarking/src/lib.rs

```diff
@@ -381,7 +374,7 @@ benchmarks! {
 		);
 	}
 
-	on_initialize {
+	on_runtime_upgrade {
```

Comment:
Note that we should remove this benchmark later once the migration is done. Perhaps document it somewhere.

Reply:
Also, I would be fine with simply returning `max_block` weight for a one-off upgrade.

Reply:
ummm ... besides: you are never using `T::WeightInfo:: on_runtime_upgrade` :P remove this please.

Reply:
So is not necessary to test the `on_runtime_upgrade`? Should I remove this benchmark completely?

Reply:
> So is not necessary to test the on_runtime_upgrade? Should I remove this benchmark completely?


It is necessary to _test_ it, but not necessary to _benchmark_ it, thus yes you can remove the benchmark.

Reply:
I knew you would say that, but I wish you didn't 😄

Reply:
In that case I need to expose the `Offense::set_deferred_offences` again.

Reply:
No you don't -- if you have a storage item for deferred offences, you can directly write to it. You don't necessarily need to use this helper function. Something like `deprecated::DeferredOffences::<T>::put(stuff)`

Reply:
Note that the tests for migration should live in the migration module.

Reply:
You are right! will do